src/Reflection/ReflectionClass.php

```diff
@@ -338,20 +338,33 @@ private function getMethodsIndexedByName() : array
     /**
      * Fetch an array of all methods for this class.
      *
+     * @param ?int $filter
      * @return ReflectionMethod[]
      */
-    public function getMethods() : array
+    public function getMethods(?int $filter = null) : array
     {
-        return array_values($this->getMethodsIndexedByName());
+        if (null === $filter) {
```

Comment:
@Ocramius Can not agree with you that calling special functions are slow. Because you missing the presence of Optimizer in PHP. So if you will use `\is_null($something);` then it won't be function call at all, because PHP optimizer will inline it as simple type check.

To prove it, check https://3v4l.org/Ivbid/vld#output to see `TYPE_CHECK` opcode for that. And it's very fast 😄  But without leading slash it will be slow function call: https://3v4l.org/asV6T/vld#output